### PR TITLE
Implement basic enumerator features

### PR DESCRIPTION
**Pull request by smithjessk**
- Add enumerator creation from a coroutine and a coroutine instance. Use ScalaMeter to assert that these functions don't cause boxing.
- Add `next`, `hasNext` methods to `Enumerator` class. Test this functionality with unit tests.
- Add ScalaMeter dependency to `coroutines-extra` subproject
- Remove unnecessary import from Coroutine.scala.

### CODE REVIEW

File: coroutines-extra/src/main/scala/org/coroutines/extra/Enumerator.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+/** Ignores and does nothing with the return value of the coroutine. This makes
+ *  specialization simpler and also makes it more straightforward for the user to
+ *  create an `Enumerator` from a `Coroutine`.
+ *
+ *  Takes a `Coroutine.Instance` over a `Coroutine` so that both the constructor is
+ *  more general and so that an enumerator can be built from an in-progress coroutine.
+ */
+class Enumerator[@specialized Y](instance: Coroutine.Instance[Y, _]) {
```
Comment (axel22) [inline coroutines-extra/src/main/scala/org/coroutines/extra/Enumerator.scala]: We should restrict specialization to `Int, Long, Double`.

File: coroutines-extra/src/main/scala/org/coroutines/extra/Enumerator.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    _hasNext = instance.pull
+    result
+  }
+}
+
+object Enumerator {
+  def apply[Y](c: Coroutine.Instance[Y, _]) = new Enumerator(c.snapshot)
+
+  def apply[Y](c: Coroutine._0[Y, _]) = new Enumerator(call(c()))
+}
```
Comment (axel22) [inline coroutines-extra/src/main/scala/org/coroutines/extra/Enumerator.scala]: Need a newline here.

File: coroutines-extra/src/main/scala/org/coroutines/extra/Enumerator.scala [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+   *
+   *  @return The result of `instance.value` after the previous call to `instance.pull`
+   */
+  def next(): Y = {
+    val result = instance.value
+    _hasNext = instance.pull
+    result
+  }
+}
+
```
Comment (axel22) [inline coroutines-extra/src/main/scala/org/coroutines/extra/Enumerator.scala]: Need 2 newlines between top-level definitions.

File: coroutines-extra/src/main/scala/org/coroutines/extra/Enumerator.scala [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+object Enumerator {
+  def apply[Y](c: Coroutine.Instance[Y, _]) = new Enumerator(c.snapshot)
+
+  def apply[Y](c: Coroutine._0[Y, _]) = new Enumerator(call(c()))
```
Comment (axel22) [inline coroutines-extra/src/main/scala/org/coroutines/extra/Enumerator.scala]: Would be nice to, in addition, to have an enumerator macro convenience wrapper that creates an enumerator directly from a block with `yieldval` statements (same as was done for `async`).

E.g.:

```scala
Enumerator {
  var i = 0
  while (i < 5) {
    yieldval(i)
    i += 1
  }
}
```

**Comment by axel22** [comment]
This LGTM, I will merge this!

Please address the few comments that I left in a separate PR.

File: coroutines-extra/src/main/scala/org/coroutines/extra/Enumerator.scala (hunk 2 above)
Comment (smithjessk) [inline coroutines-extra/src/main/scala/org/coroutines/extra/Enumerator.scala]: Which line?

**Comment by smithjessk** [comment]
Done! #35 addresses these issues.

**Pull request merged**